app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -57,56 +85,54 @@ def _validate_tifffile(  # noqa: C901
 
     # Checks if the image description exists,
     # if so, ensure there's no DICOM or XML data
-    try:
-        image_description = str(tags["ImageDescription"].value).lower()
-        for forbidden in forbidden_description_tags:
-            if forbidden in image_description:
-                raise ValidationError(
-                    "Image contains unauthorized information"
-                )
-    except KeyError:
-        pass
+    image_description = str(
+        get_tag_value(tags, "ImageDescription", False)
+    ).lower()
+    for forbidden in forbidden_description_tags:
+        if forbidden in image_description:
+            raise ValidationError("Image contains unauthorized information")
 
     # Fails if the image doesn't have all required tile tags
-    if not all(tag in tags for tag in required_tile_tags):
-        raise ValidationError("Image has incomplete tile information")
+    for tag in required_tile_tags:
+        get_tag_value(tags, tag, True)
 
     # Fails if the image only has a single resolution page
     resolution_levels = len(pages)
     if resolution_levels == 1:
         raise ValidationError("Image only has a single resolution level")
 
     # Fails if the image doesn't have the chunky format
-    if str(tags["PlanarConfiguration"].value) != "PLANARCONFIG.CONTIG":
+    if (
+        str(get_tag_value(tags, "PlanarConfiguration"))
+        != "PLANARCONFIG.CONTIG"
+    ):
         raise ValidationError(
             "Image planar configuration isn't configured as 'Chunky' format"
         )
 
     # Fails if the color space isn't supported
-    try:
-        color_space = get_color_space(
-            str(tags["PhotometricInterpretation"].value)
-        )
-    except KeyError:
-        raise ValidationError("Image lacks color space information")
+    color_space = get_color_space(
+        str(get_tag_value(tags, "PhotometricInterpretation"))
+    )
 
     # Fails if the amount of bytes per sample doesn't correspond to the
     # colour space
-    tif_color_channels = tags["SamplesPerPixel"].value
+    tif_color_channels = get_tag_value(tags, "SamplesPerPixel")
     if Image.COLOR_SPACE_COMPONENTS[color_space] != tif_color_channels:
         raise ValidationError("Image contains invalid amount of channels.")
 
-    try:
-        image_width = tags["ImageWidth"].value
-        image_height = tags["ImageLength"].value
-    except KeyError:
-        raise ValidationError("Missing tags in tiff file")
+    pixel_width = calculate_pixel_size(tags, "XResolution")
+    pixel_height = calculate_pixel_size(tags, "YResolution")
```

Comment:
Do you also have ZResolution in tiff (eg, for the thickness of the slide)

Reply:
Not that I found, I'll check with Geert

Reply:
Nope, defaults to 1